.github/workflows/regression.yml

```diff
@@ -68,7 +68,50 @@ jobs:
           name: ${{ matrix.config_set }}
           path: install_bin.tgz
       - name: Post Cleanup
-        uses: docker://ghcr.io/su2code/su2/build-su2:230225-2136
+        uses: docker://ghcr.io/su2code/su2/build-su2:230704-1323
+        with:
+          entrypoint: /bin/rm
+          args: -rf install install_bin.tgz src ccache ${{ matrix.config_set }}
+  build_tsan:
+    name: Build SU2 (tsan)
+    if: inputs.runner != 'ARM64'
+    strategy:
+      fail-fast: false
+      matrix:
+        config_set: [BaseOMP-tsan, ReverseOMP-tsan] #ForwardOMP-tsan
+        include:
+          - config_set: BaseOMP-tsan
+            flags: '--buildtype=debugoptimized -Dwith-omp=true -Denable-mixedprec=true -Denable-tecio=false --warnlevel=3'
+          - config_set: ReverseOMP-tsan
+            flags: '--buildtype=debugoptimized -Denable-autodiff=true -Denable-normal=false -Dwith-omp=true -Denable-mixedprec=true -Denable-tecio=false --warnlevel=3'
```

Comment:
Optimized debug builds have reasonable runtime and stack traces that are still useful if a data race is detected. This is in line with the [recommendations](https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual).

I had to remove the `-werror`, there is warning about control reaching the end of the non-void function `CMultizoneDriver::Monitor()`. IIRC a fix for this is pending somewhere?

Reply:
Yep in #2011

Reply:
The `-werror` can be readded there :+1: